File: src/main/java/com/dv/telegram/data/WikiBotCommandsParser.java
package com.dv.telegram.data;

import com.dv.telegram.google.RowData;
import com.dv.telegram.google.SheetData;
import com.dv.telegram.google.WikiBotGoogleSheet;
import lombok.extern.log4j.Log4j2;

import java.util.ArrayList;
import java.util.List;

@Log4j2
public class WikiBotCommandsParser implements SheetDataParser<WikiBotCommandData> {

    @Override
    public SheetData getSheetData(WikiBotGoogleSheet sheet) {
        return sheet.getCommandsSheet();
    }

    @Override
    public List<WikiBotCommandData> parse(List<RowData> rows) {
        List<WikiBotCommandData> commands = new ArrayList<>();

        int rowNum = 1;

        for (RowData row : rows) {
            String answer = row.getCellOrBlank(0);
            String wordsString = row.getCellOrBlank(1);

            List<String> words = DataUtils.parseWords(wordsString);

            WikiBotCommandData command = new WikiBotCommandData(answer, wordsString, words);
            commands.add(command);

            log.info("Row {}: / {} / {}", rowNum, answer, words);
            rowNum++;
        }

        log.info("Total {} commands parsed.", commands.size());

        return commands;
    }
}

File: src/main/kotlin/com/dv/telegram/data/WikiBotCommandsParser.kt
package com.dv.telegram.data

import com.dv.telegram.google.RowData
import com.dv.telegram.google.WikiBotGoogleSheet
import com.dv.telegram.tabs.TabFormat
import org.apache.logging.log4j.kotlin.Logging

class WikiBotCommandsParser : SheetDataParser<WikiBotCommandData>, Logging {

    override fun getSheetData(sheet: WikiBotGoogleSheet) = sheet.commandsSheet

    override fun getTabFormat(): TabFormat {
        return TabFormat.COMMANDS
    }

    override fun parse(rows: List<RowData>): List<WikiBotCommandData> {
        val commands = mutableListOf<WikiBotCommandData>()

        var rowNum = 1

        for (row in rows) {
            val answer = row.getCellOrBlank(0)
            val wordsString = row.getCellOrBlank(1)
            val words = DataUtils.parseWords(wordsString)

            val command = WikiBotCommandData(answer, wordsString, words)
            commands.add(command)

            logger.info("Row $rowNum: / $answer / $words")
            rowNum++
        }

        logger.info("Total ${commands.size} commands parsed.")

        return commands
    }
}
